docs/src/pages/customization/density/density.md

```diff
@@ -0,0 +1,102 @@
+# Density
+
+<p class="description">We explain how to apply density to Material-UI components</p>
+
+## Applying density
+
+We won't cover possible use cases, or considerations for using density in your application.
+The Material design guidelines have a [comprehensive guide](https://material.io/design/layout/applying-density.html#typographic-density) covering these topics in more detail.
+
+## Implementing density
+
+Higher density can be applied to some components via props. The component pages
+have at least one example using the respective component with higher density applied.
+
+Depending on the component, density is applied either via lower spacing, or simply by
+reducing the size.
+
+The following components have props applying higher density:
+
+- [Button](/api/button)
+- [Fab](/api/fab)
+- [FilledInput](/api/filled-input)
+- [FormControl](/api/form-control)
+- [FormHelperText](/api/form-helper-text)
+- [IconButton](/api/icon-button)
+- [InputBase](/api/input-base)
+- [InputLabel](/api/input-label)
+- [ListItem](/api/list-item)
+- [OutlinedInput](/api/outlined-input)
+- [Table](/api/table)
+- [TextField](/api/text-field)
+- [Toolbar](/api/toolbar)
+
+## Explore theme density
+
+This tool allows you to apply density via spacing and component props. You can browse
+around and see how this applies to the overall feel of Material-UI components.
+
+If you enable high density we apply a custom theme to our docs. This theme is only
+for demonstration purposes. You _should not_ apply this theme to your whole application
+as this might impact user experience negatively. The [Material design guidelines](https://material.io/design/layout/applying-density.html#typographic-density) has examples
```

Comment:
```suggestion
as this might negatively impact user experience. The [Material design guidelines](https://material.io/design/layout/applying-density.html#typographic-density) has examples
```